ersilia/core/tracking.py

```diff
@@ -81,53 +85,65 @@ def log_files_metrics(file):
                 if misc_error_flag:
                     errors[error_name] += 1
 
-        write_persistent_file(f"Error count: {error_count}")
+        write_persistent_file(f"Error count: {error_count}", model_id)
         if len(errors) > 0:
-            write_persistent_file(f"Breakdown by error types:")
+            write_persistent_file(f"Breakdown by error types:", model_id)
             for error in errors:
-                write_persistent_file(f"{error}: {errors[error]}")
-        write_persistent_file(f"Warning count: {warning_count}")
+                write_persistent_file(f"{error}: {errors[error]}", model_id)
+        write_persistent_file(f"Warning count: {warning_count}", model_id)
     except (IsADirectoryError, FileNotFoundError):
         logging.warning("Unable to calculate metrics for log file: log file not found")
-        
 
-def open_persistent_file(model_id):
+
+
+def create_persistent_file(model_id):
+
     """
-    Opens a new persistent file, specifically for a run of model_id
+    Create the persistent path file 
     :param model_id: The currently running model
     """
-    with open(PERSISTENT_FILE_PATH, "w") as f:
+
+    persistent_file_dir = os.path.join(EOS, ERSILIA_RUNS_FOLDER, "session", model_id)
+    os.makedirs(persistent_file_dir, exist_ok=True)  
+    file_name = os.path.join(persistent_file_dir, "current_session.txt")
+    
+    with open(file_name, "w") as f:
         f.write("Session started for model: {0}\n".format(model_id))
+    
+    return file_name
+
+    
         
-        
-def write_persistent_file(contents):
+def write_persistent_file(contents, model_id):
+
     """
     Writes contents to the current persistent file. Only writes if the file actually exists.
     :param contents: The contents to write to the file.
     """
-
-    # Only write to file if it already exists (we're meant to be tracking this run)
-    if os.path.isfile(PERSISTENT_FILE_PATH):
-        with open(PERSISTENT_FILE_PATH, "a") as f:
+    
+    file_name = create_persistent_file(model_id)
```

Comment:
I'm sorry I don't understand why we need to create the file before writing to it or closing it every time? This write function is supposed to write in append mode only when the file already exists. A better approach would be to have a different `check_file_exists` method if that is what we are trying to ensure here.